docs/training-apis/experiment-config.txt

```diff
@@ -473,27 +473,45 @@ example:
 
    hyperparameters:
      global_batch_size: 64
-     optimizer:
-       type: categorical
-       vals:
-         - SGD
-         - Adam
-         - RMSprop
+     optimizer_config:
+       optimizer:
+         type: categorical
+         vals:
+           - SGD
+           - Adam
+           - RMSprop
+       learning_rate:
+         type: log
+         minval: -5.0
+         maxval: 1.0
+         base: 10.0
+     num_layers:
+       type: int
+       minval: 1
+       maxval: 3
      layer1_dropout:
        type: double
        minval: 0.2
        maxval: 0.5
-     learning_rate:
-       type: log
-       minval: -5.0
-       maxval: 1.0
-       base: 10.0
-
-This configuration defines four hyperparameters: ``global_batch_size``, ``optimizer``,
-``layer1_dropout``, and ``learning_rate``. ``global_batch_size`` is set to a constant value; the
-other hyperparameters can take on a range of possible values. A hyperparameter's range is configured
-by the ``type`` field of the map; it must be one of ``categorical``, ``double``, ``int``, or
-``log``. More details on these types are given below.
+
+This configuration defines the following hyperparameters:
```

Comment:
I'd rather see this as a table with the following columns: 
field name | type | description (from what follows the bullet list)